### PR TITLE
feat(chat): add the 'Add the agent to My workspace to continue' button (Issues #2364, #2454)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

add the 'Add the model to My workspace to continue' button

Issues:

- Issue #2364
- Issue #2454 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — DISMISSED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by nartovm** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/store/models/models.reducers.ts [hunk 1]
```diff
@@ -253,6 +259,13 @@ const selectIsModelsLoaded = createSelector([rootSelector], (state) => {
   return state.status === UploadStatus.LOADED;
 });
 
+const selectIsInstalledModelsInitialized = createSelector(
+  [rootSelector],
+  (state) => {
+    return state.installedModelsStatus !== UploadStatus.UNINITIALIZED;
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.reducers.ts]: need to exclude `LOADING` also
```suggestion
    return state.installedModelsStatus === UploadStatus.LOADED;
```

File: apps/chat/src/components/Chat/ChatInput/ChatExternalControls.tsx [hunk 2]
```diff
@@ -66,7 +67,9 @@ export default function ChatExternalControls({
           <span className="text-secondary">
             <IconCopy width={18} height={18} />
           </span>
-          {t('Duplicate the conversation to be able to edit it')}
+          {conversationsToDuplicate.length > 1
+            ? t('Duplicate conversations to be able to edit it')
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatInput/ChatExternalControls.tsx]: difference only with `s`
```suggestion
            ? t('Duplicate the conversations to be able to edit it')
```

File: apps/chat/src/components/Chat/ChatInput/AddModelsControl.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    <div className="flex justify-center">
+      <div className="relative mx-2 mb-2 flex w-full flex-row items-center justify-center gap-3 md:mx-4 md:mb-0 md:last:mb-6 lg:mx-auto lg:w-[768px] lg:max-w-3xl">
+        <button
+          onClick={handleInstallModels}
+          className="button inset-x-0 !-top-10 mx-auto flex w-fit items-center gap-2 border-primary bg-layer-2 p-3 hover:bg-layer-4"
+          data-qa="add-model-to-workspace"
+        >
+          <IconPlayerPlay size={18} />
+          {modelIdsToInstall.length > 1
+            ? t('Add agents to My workspace to continue')
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatInput/AddModelsControl.tsx]: difference only with `s`
```suggestion
            ? t('Add the agents to My workspace to continue')
```

File: apps/chat/src/store/models/models.epics.ts [hunk 4]
```diff
@@ -317,8 +317,8 @@ const addInstalledModelsEpic: AppEpic = (action$, state$) =>
                   of(
                     UIActions.showSuccessToast(
                       payload.references.length > 1
-                        ? translate('Models added to my workspace')
-                        : translate('The model added to my workspace'),
+                        ? translate('Agents added to my workspace')
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts]: I don't like to use new term `agent`
```suggestion
                        ? translate('The agents added to my workspace')
```

File: apps/chat/src/components/Chat/ChatInput/ChatInputControls.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        showScrollDownButton={showScrollDownButton}
+        onScrollDown={onScrollDown}
+      />
+    );
+  }
+
+  return null;
+};
+
+export default ChatInputControls;
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatInput/ChatInputControls.tsx]: do we need default export?

File: apps/chat/src/components/Chat/Chat.tsx [hunk 6]
```diff
@@ -965,9 +971,12 @@ export function Chat() {
...
   if (
-    !areSelectedConversationsLoaded &&
-    (!selectedConversations.length ||
-      selectedConversations.some((conv) => conv.status !== UploadStatus.LOADED))
+    (!areSelectedConversationsLoaded &&
+      (!selectedConversations.length ||
+        selectedConversations.some(
+          (conv) => conv.status !== UploadStatus.LOADED,
+        ))) ||
+    !isInstalledModelsInitialized
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Chat.tsx]:
```suggestion
    !areSelectedConversationsLoaded ||
    selectedConversations.some(
      (conv) => conv.status !== UploadStatus.LOADED,
    ) ||
    !isInstalledModelsInitialized
```

File: apps/chat/src/store/models/models.reducers.ts (hunk 1 above)
Comment (Alexander-Kezik) [reply apps/chat/src/store/models/models.reducers.ts]: so we will have global loading each time when we open settings. I don't think that is good idea

File: apps/chat/src/components/Chat/ChatInput/ChatExternalControls.tsx [hunk 7]
```diff
@@ -66,7 +67,9 @@ export default function ChatExternalControls({
           <span className="text-secondary">
             <IconCopy width={18} height={18} />
           </span>
-          {t('Duplicate the conversation to be able to edit it')}
+          {t(
+            `Duplicate the ${conversationsToDuplicate.length > 1 ? 'conversations' : 'conversation'} to be able to edit it`,
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/ChatInput/ChatExternalControls.tsx]: I suggest to use such logic for all plural/single messages
```suggestion
            `Duplicate the conversation${conversationsToDuplicate.length > 1 ? 's' : ''} to be able to edit it`,
```

Comment (IlyaBondar) [reply apps/chat/src/components/Chat/ChatInput/ChatExternalControls.tsx]: we can add such method:
```js
export const plurar = (term: string, items: unknown[]) =>
  `${term}${items.length > 1 ? 's' : ''}`;
```

File: apps/chat/src/store/models/models.epics.ts [hunk 8]
```diff
@@ -317,8 +317,8 @@ const addInstalledModelsEpic: AppEpic = (action$, state$) =>
                   of(
                     UIActions.showSuccessToast(
                       payload.references.length > 1
-                        ? translate('Models added to my workspace')
-                        : translate('The model added to my workspace'),
+                        ? translate('The agents added to my workspace')
+                        : translate('The agent added to my workspace'),
```
Comment (IlyaBondar) [inline apps/chat/src/store/models/models.epics.ts]:
```
translate(`The agent${payload.references.length > 1 ? 's' : ''} added to my workspace`)
```

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — DISMISSED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>- **Deployment failed**: [canceled](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/531608)

**Comment by Alexander-Kezik** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/531638)

**Comment by Alexander-Kezik** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/531682)

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2440.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Review by nartovm — APPROVED** [review]

**Pull request merged**